packages/web/src/components/typography/Text.tsx

```diff
@@ -0,0 +1,90 @@
+// import cn from 'classnames'
+
+import { CSSProperties, ReactNode } from 'react'
+
+import { getCurrentThemeColors, ThemeColor } from 'utils/theme/theme'
+
+import { variantInfoMap, variantStrengthMap } from './constants'
+import {
+  TextSizeInfo,
+  TextStrengthInfo,
+  TextVariant,
+  TextSize,
+  TextStrength,
+  FontWeight
+} from './types'
+
+// TODO:
+// - Get theme colors in here
+
+export const fontWeightMap: Record<FontWeight, number> = {
+  heavy: 900,
+  bold: 700,
+  demiBold: 600,
+  medium: 500,
+  regular: 400,
+  light: 300,
+  thin: 200,
+  ultraLight: 100
+}
+
+const defaultBodyInfo: TextSizeInfo & TextStrengthInfo = {
+  tag: 'p',
+  fontSize: '16px',
+  lineHeight: '20px',
+  fontWeight: 'medium',
+  isAllCaps: false
+}
+
+export type TextProps = {
+  className?: string
+  children?: ReactNode
+  variant?: TextVariant
+  size?: TextSize
+  strength?: TextStrength
+  color?: ThemeColor
+}
+
+export const Text = ({
+  className,
+  children,
+  variant = 'body',
+  strength = 'default',
+  size = 'M',
+  color = '--text-default',
+  ...props
+}: TextProps) => {
+  const themeColors = getCurrentThemeColors()
+  const textColor = themeColors[color] ?? themeColors['--text-default']
+
+  const variantSizeInfo = variantInfoMap[variant][size] ?? {}
+  const variantStrengthInfo = variantStrengthMap[variant][strength] ?? {}
+
+  const {
+    tag: Tag,
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    fontWeight,
+    isAllCaps
+  } = {
+    ...defaultBodyInfo,
+    ...variantSizeInfo,
+    ...variantStrengthInfo
+  }
+
+  const styleObject: CSSProperties = {
+    fontWeight: fontWeightMap[fontWeight].toString(),
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    textTransform: isAllCaps ? 'uppercase' : 'inherit',
```

Comment:
is there a reason to use `isAllCaps` instead of `textTransform` as the key?

Reply:
was on the fence about this, felt weird to type out the values for a css prop, but will try it out